### ISSUE TITLE
LDAP import error: "Unsupported object type: id=0"

### ISSUE THREAD

**Issue by larrybg** [zammad/zammad#2631]
### Infos:

* Used Zammad version: 3.0.x
* Installation method (source, package, ..): yum
* Operating system: CentOS 7
* Database + version:
* Elasticsearch version: 5.6.16
* Browser + version: Chrome 75.0.3770.100

### Expected behavior:
Working LDAP integration. The application connects to the LDAP server and imports users from specified EXISTING LDAP groups.
*

### Actual behavior:
LDAP configuration is failing with error: "Unsupported object type: id=0". 
*

### Steps to reproduce the behavior:
Connect to LDAP. 
Specify user mapping from the dropdowns. 
Select **EXISTING** LDAP groups from the dropdowns and assign Roles. 
Select not to sync users without LDAP groups and click next.
After a few minutes of calculating users, error will be shown"Unsupported object type: id=0".

I've tried to set up LDAP integration using several different EXISTING groups selecting them from the dropdown list and never succeeded.
I've submitted another issue #2630 but you called it a "feature"...  LDAP integration doesn't work.
*

Yes, I'm sure this is a bug and no feature request or a general question.

**Comment by larrybg** [comment]
I’ve confirmed that groups I’m after are listed in another LDAP browser.

**Comment by MrGeneration** [comment]
Can you please share your mapping of LDAP attributes to Zammad attributes? :)

Edit: Used LDAP would be interesting too (Active Directory, OpenLDAP, ...)

**Comment by larrybg** [comment]
Sorry, I'm traveling this week, so my replies can take longer than usual.
I have made only one change once attributes were loaded - pointed company’s email to Login. Attaching the screenshot. Please let me know if this is what you have been asking
<img width="664" alt="Screen Shot 2019-06-28 at 9 17 11 PM" src="https://user-images.githubusercontent.com/20246246/60379634-352cbf00-99eb-11e9-89bf-1880669e8c5b.png">
.

**Comment by MrGeneration** [comment]
I tried to verify this, but Zammad currently won't allow me to add two identical ldap attributes to the sync:
![image](https://user-images.githubusercontent.com/6549061/60594994-708e0b80-9da6-11e9-8b5d-301ffec9e0ad.png)

After talking to a developer, this should never have worked. (and thus works as designed).

Right now I have absolutely no clue how to reproduce this. Currently I expect that you upgraded Zammad from an older version with a configured ldap configuration and now started to receive those errors.

### So, to verify this, please provide the following information:
* Zammad-Version you upgraded from
* Used LDAP-Software (MS Active Directory, OpenLDAP, ....) **plus Version**

The above requested information are mandatory for me trying to reproduce this.

Another interesting question for me would be if this has worked before and if not, how you configured the above shown ldap mapping to Zammad. If this worked before, you might need to update the ldap-config (but again, it might be because of a migration from an older version, even though your posted configuration should never have worked)

**Comment by larrybg** [comment]
Hi! We've never used Zammad before, so this is a new installation (no upgrade).
I've also asked about LDAP software & version, but afraid will get an answer only after the Independence Day (4th of July).

**Comment by larrybg** [comment]
are there any updates on this issue? 
I’ve used an LDAP Browser I’ve downloaded from here: www.ldapadministrator.com and able to connect with the same settings. So I can confirm that our ldap works just fine. I also can filter/search my custom ldap groups using that browser, something that fails with Zammad. I’ve tried to copy the group's settings from LDAP browser to Zammad and it didn’t work either.

**Comment by thorsteneckel** [comment]
Can you please reproduce it on one of the [free hosted instances](https://zammad.com/getting-started) and let us know the name?

**Comment by larrybg** [comment]
I'm afraid this is impossible as your instances are outside our company's network.

**Comment by MrGeneration** [comment]
Sorry, but reproducing this issue then becomes impossible for us. :-(

**Comment by larrybg** [comment]
I've made one step forward - changed the "User filter" to the correct value and now I can pass this error, but cannot map "LDAP Groups" to Zammad Roles. What value is correct to use in Roles: `cn=ZammadAdmins,ou=memberlist,ou=compgroups,o=comp.com` or `cn=ZammadAdmins,ou=metadata,ou=compgroups,o=comp.com`?
Actually, none of these values are shown in the dropdown, any ideas why?

**Comment by MrGeneration** [comment]
Moving the discussion part to the correspodending community post to keep the noise on the tracker as low as possible:
https://community.zammad.org/t/ldap-integration-is-failing/2704/4

We'll update the issue accordingly if we found the issue.

**Comment by MrGeneration** [comment]
I'm closing this issue because we've lost track within the community. 
I currently believe that this is not a specific issue to `Tivoli Directory Server` (which was used here) but possibly with the data structure which makes this an technical question possibly.
